### PR TITLE
add is_resize_mask cfg in inst seg

### PR DESCRIPTION
**Pull request by ruisv**
## Modification

Add is_resize_mask configuration option in instance segmentation.
Support output cropped mask or mask consistent with input image shape.

### CODE REVIEW

File: csrc/mmdeploy/codebase/mmdet/instance_segmentation.cpp [hunk 1]
```diff
@@ -165,6 +169,7 @@ class ResizeInstanceMask : public ResizeBBox {
   operation::Managed<operation::WarpAffine> warp_affine_;
   float mask_thr_binary_{.5f};
   bool is_rcnn_{true};
+  bool is_resize_mask_{true};
```
Comment (irexyc) [inline csrc/mmdeploy/codebase/mmdet/instance_segmentation.cpp:172]:
```suggestion
  bool is_resize_mask_{false};
```

I think it's better to make it false as default for backward compatible.

File: mmdeploy/codebase/mmdet/deploy/object_detection.py [hunk 2]
```diff
@@ -316,6 +316,7 @@ def get_postprocess(self, *args, **kwargs) -> Dict:
                 params['mask_thr_binary'] = params['rcnn']['mask_thr_binary']
         if 'mask_thr_binary' in params:
             type = 'ResizeInstanceMask'  # for instance-seg
+            params['is_resize_mask'] = True  # resize and crop mask default
```
Comment (irexyc) [inline mmdeploy/codebase/mmdet/deploy/object_detection.py:319]:
```suggestion
            params['is_resize_mask'] = False  # resize and crop mask default
```

same reason as above

File: csrc/mmdeploy/codebase/mmdet/instance_segmentation.cpp (hunk 1 above)
Comment (ruisv) [reply csrc/mmdeploy/codebase/mmdet/instance_segmentation.cpp:172]: 👌

File: mmdeploy/codebase/mmdet/deploy/object_detection.py (hunk 2 above)
Comment (ruisv) [reply mmdeploy/codebase/mmdet/deploy/object_detection.py:319]: 👌

**Review by irexyc — APPROVED** [review]
Thanks for your PR.

**Comment by RunningLeon** [comment]
@ruisv hi, could kindly fix the [lint](https://github.com/open-mmlab/mmdeploy/actions/runs/5421499906/jobs/9857133078?pr=2229) error

**Comment by ruisv** [comment]
@RunningLeon  Hi, this is my first PR, could u give me some suggestions. Thank u.

**Comment by ruisv** [comment]
@RunningLeon  remove this line ?
image

**Review by irexyc — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @ruisv,

We'd like to express our appreciation for your valuable contributions to the mmdeploy. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

**Pull request merged**